### ISSUE TITLE
Default Radio values set incorrectly due to non-strict type checking

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1367]
I found a problem with default form element values when the element type is "Boolean", or "List (integer)" and the widget is "Check boxes/radio buttons" (single value, so set to radio). 

Here is an example of two fields, each has NO DEFAULT SET, yet the interface shows a default of the zero/false value in both cases, and the source code shows `checked="checked"` has been added. 

<img width="551" alt="screen shot 2017-04-30 at 1 16 22 pm" src="https://cloud.githubusercontent.com/assets/397895/25567639/4f2b5c84-2da7-11e7-806c-be6903d71883.png">

There's a check in `theme_radio` that compares two values ignoring type, so FALSE will match 0:

``` php
$element['#value'] == $element['#return_value']
```

This should be corrected to something like...

``` php
$element['#value'] === $element['#return_value']
```

---
PR https://github.com/backdrop/backdrop/pull/1126

**Comment by quicksketch** [comment]
From the PR at https://github.com/backdrop/backdrop/issues/1126, this doesn't look so novice any more.

The PR fix looks good to me, but I think we need additional tests to ensure 0/FALSE/`''` comparisons work as expected after this change.

**Comment by jenlampton** [comment]
I added some new tests, but I'm not entirely sure they are testing what I think they are testing.

**Comment by jenlampton** [comment]
Okay, I revisited this bug in the light of day, and I was finally able to add tests that fail before the fix, and pass afterward. Setting milestone to 1.6.4.

**Comment by jenlampton** [comment]
I just played with this again, and I can't get it to fail consistently. I'd like confirmation from someone else that this is a real, verifiable problem before we commit the fix (removing milestone).

**Comment by Dinornis** [comment]
When was this bug introduced?
I am not able to replicate this in 1.6.0

**Comment by jenlampton** [comment]
I'm not sure, I noticed it for the first time on Nov 25, 2015.

**Comment by Dinornis** [comment]
Must have been fixed since? I am not able to reproduce in 1.7.x-dev

**Comment by jenlampton** [comment]
Okay, well let's close for now. If someone has this problem again we can re-open or file a new issue. Thanks for testing!
